Update oracle.md (#426)
solana-foundation/developer-content: content/courses/connecting-to-offchain-data/oracles.md

To use Switchboard oracles and incorporate off-chain data into a Solana program,
you first have to find a feed that provides the data you need. Switchboard feeds
are public and there are many

> [already available that you can choose from](https://app.switchboard.xyz/solana/devnet).

Reviewer: ```suggestion
[feeds to can choose from](https://app.switchboard.xyz/solana/devnet).
```

Author: done

it’s the account you expect it to be.

In Anchor, if you deserialize the account to the `AggregatorAccountData` type
from the `switchboard_solana` crate, Anchor checks that the account is owned by
the Switchboard program. If your program expects that only a specific data feed
will be passed in the instruction, then you can also verify that the public key
of the account passed in matches what it should be. One way to do this is to

> hard code the address in the program somewhere and use account constraints to

Reviewer: ```suggestion
hard code the address in the program somewhere and use [account constraints](https://www.anchor-lang.com/docs/account-constraints) to
```

it’s the account you expect it to be.

In Anchor, if you deserialize the account to the `AggregatorAccountData` type
from the `switchboard_solana` crate, Anchor checks that the account is owned by
the Switchboard program. If your program expects that only a specific data feed
will be passed in the instruction, then you can also verify that the public key
of the account passed in matches what it should be. One way to do this is to
hard code the address in the program somewhere and use
[account constraints](https://www.anchor-lang.com/docs/account-constraints) to
verify the address passed in matches what is expected.

```rust
use {
  anchor_lang::prelude::*,
  solana_program::{pubkey, pubkey::Pubkey},

> 	switchboard_solana::{AggregatorAccountData},

Reviewer: Can you also format the rust (4 spaces is what rustfmt uses) for this and the other repo? See https://github.com/solana-foundation/developer-content/blob/main/CONTRIBUTING.md#rust--anchor

Author: sure

    use super::*;

>     pub fn deposit(ctx: Context<Deposit>, escrow_amt: u64, unlock_price: f64) -> Result<()> {

Reviewer: Why a float? This seems very odd - computers rarely use floats for financial operations, rather using ints for minor units, due to decimal math issues.

Author: @mikemaccana  need suggestion : will it be fine to use u64  for unlock_price ?

Reviewer: It certainly seems safer - 

 - unsigned (nothing has a negative price)
 - whole numbers rather than decimals
 
 Where did the idea to use f64 come from?

Reviewer: (Just a note that I wrote a comment about renaming withdraw_handler, but that's how Anchor does it, so i deleted the comment)

Author: i think f64 is used as sol price will be a floating point ex $21.53

Reviewer: SOL price will be *shown* as a floating point, but *everything* on Solana is done in whole amounts. Eg, transferring SOL is done in lamports. transferring USDC is done in 2 ^ 6 minor USDC units.

Reviewer: This is the same in TradFI too BTW - prices are in cents, pence, etc. although shown in dollars / GBP etc.

>         deposit_handler(ctx, escrow_amt, unlock_price)

Reviewer: The old code has a couple of unnecessary wrapper functions. Also, technically, the function that receives the incoming instruction first is the instruction handler.

Reviewer: Also `amt` is a bad variable name, see https://github.com/solana-foundation/developer-content/blob/main/CONTRIBUTING.md#code

Reviewer: Same with `val`

      seeds = [ESCROW_SEED, user.key().as_ref()],
      bump,
      payer = user,

>       space =  EscrowState::INIT_SPACE + 8

Reviewer: make a usize for ANCHOR_DISCRIMINATOR and set it to 8.

Reviewer: Oh also call EscrowState just Escrow. It's a PDA, we know it stores state.

Author: ```console
#[account]
[derive(InitSpace)]
pub struct EscrowState {
    pub unlock_price: f64,
    pub escrow_amount: u64,
}
````
 @mikemaccana  it is a struct in state.rs should i rename it

Reviewer: Yes. state.rs already tells us that it's state.

Author: done

} from "@switchboard-xyz/solana.js";
import { assert } from "chai";

> export const solUsdSwitchboardFeed = new anchor.web3.PublicKey(

Reviewer: Also add a comment

```
// See https:// (somewhere this address is published)
```

Author: done

  SwitchboardProgram,
} from "@switchboard-xyz/solana.js";
import { assert } from "chai";

> //for SOL_USD  See https://app.switchboard.xyz/solana/devnet/feed/GvDMxPzN1sCj7L26YDK2HnMRXEQmQ2aemov8YBtPS7vR

Reviewer: ```suggestion
// For SOL_USD - see https://app.switchboard.xyz/solana/devnet/feed/GvDMxPzN1sCj7L26YDK2HnMRXEQmQ2aemov8YBtPS7vR
```

Author: done
